File: tests/src/nl/t64/game/rpg/components/party/SkillItemIdTest.java
package nl.t64.game.rpg.components.party;

import org.assertj.core.api.Assertions;
import org.junit.jupiter.api.Test;


class SkillItemIdTest {

    @Test
    void test() {
        Assertions.assertThat(SkillItemId.from("diplomat")).containsSame(SkillItemId.DIPLOMAT);
        Assertions.assertThat(SkillItemId.from("pipo")).isEmpty();
    }

}

File: tests/src/nl/t64/game/rpg/components/party/SkillItemIdTest.kt
package nl.t64.game.rpg.components.party

import nl.t64.game.rpg.components.party.skills.SkillItemId
import nl.t64.game.rpg.components.party.skills.toSkillItemId
import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatExceptionOfType
import org.junit.jupiter.api.Test


internal class SkillItemIdTest {

    @Test
    fun test() {
        assertThat("diplomat".toSkillItemId()).isSameAs(SkillItemId.DIPLOMAT)
        assertThatExceptionOfType(NoSuchElementException::class.java).isThrownBy {
            "pipo".toSkillItemId()
        }
    }

}
